src/middleware.rs

```diff
@@ -16,76 +16,46 @@
 //! ```
 //!
 //! A request first travels through all BeforeMiddleware, then a Response is generated
-//! by the Handler, which can be an arbitrary nesting of AroundMiddleware, then all AfterMiddleware
-//! are called with both the Request and Response. After all AfterMiddleware have been fired,
-//! the response is written back to the client.
+//! by the Handler, which can be an arbitrary nesting of AroundMiddleware, then all
+//! AfterMiddleware are called with both the Request and Response. After all AfterMiddleware
+//! have been fired, the response is written back to the client.
 //!
-//! Not all too surprising. Of note is that AfterMiddleware and BeforeMiddleware
-//! are completely separate, unlike in many other web frameworks.
+//! Iron's error handling system is pragmatic and focuses on tracking two pieces
+//! of information for error receivers (other middleware):
+//!   - The cause of the error
+//!   - The result (what to do about) the error.
 //!
-//! Iron's flexibility comes into play with its novel approach to error handling, which encourages
-//! not just reporting errors, but *handling* them. An error is not meant to be fatal in Iron, but
-//! is instead likely to be handled by downstream Middleware. Middleware authors should keep this
-//! in mind when designing their APIs.
+//! The cause of the error is represented simply by the error itself, and the result
+//! of the error, representing the action to take in response to the error, is a complete
+//! Response, which will be sent at the end of the error flow.
 //!
-//! Iron's error propagation and handling scheme is inspired by two primary rules:
+//! When an error is thrown in Iron by any middleware or handler returning an `Err`
+//! variant with an `IronError`, the flow of the Request switches to the error flow,
+//! which proceeds to just call the `catch` method of middleware and sidesteps the
+//! `Handler` entirely, since there is already a Response in the error.
 //!
-//!   * Errors should persist and be propagated along the same route as successes until they are
-//!     handled.
+//! Once a Request has entered the error flow, it is impossible to exit. It is therefore
+//! impossible to fully "handle" an error and return to the normal flow, since there is
+//! often no sensical place to return to and intermingling the two flows results in
```

Comment:
Nitpick: "sensible" instead of "sensical"? I love nonsensical as a word, but sensical doesn't seem to be a word, or at least not a recognised/common word.